### PR TITLE
New Features!

### PR DESCRIPTION
**Pull request by Roosterseatyou**
Adds support for Player names, Animal Names, and Join Messages.

All works as intended. I was going to do Item names, but it proved more annoying than anticipated and I'll hopefully work on it later.

### CODE REVIEW

**Review by Roosterseatyou — COMMENTED** [review]
I'd prefer we didn't do this, as Paper's Adventure API is a lot better than Spigot's (IMO) and this just seems kinda unnecessary since everything was separated nicely and it *shouldn't* have been leading to any performance issues. Is there a specific reason why you wanted it to be done this way?

**Comment by treecaptcha** [comment]
@Roosterseatyou 
I was just trying to make the code simpler by inheriting duplicate listeners
I will revert that commit now :)

File: src/main/java/ml/treecaptcha/uwuify/paper/PaperUwuHandler.java [hunk 1]
```diff
@@ -37,17 +32,32 @@ public void onSignChange(org.bukkit.event.block.SignChangeEvent e) {
...
-            meta.title(Component.text(Uwuifier.uwuify(serializer.serialize(meta.title()))));
+            meta.title(Component.text(Uwuifier.uwuify(AdventureChat.twoString(meta.title()))));
         }
         e.setNewBookMeta(meta);
     }
+
+    @EventHandler
+    public void onEntityName(PlayerNameEntityEvent e) {
+        if(!Uwuify.ANIMALS_UWUIFY) return;
+        e.setName(Component.text(Uwuifier.uwuifyMessage(AdventureChat.twoString(e.getName()))));
```
[the remark this answers is not in the document]
Comment (treecaptcha) [reply src/main/java/ml/treecaptcha/uwuify/paper/PaperUwuHandler.java]: fixed!

**Review by treecaptcha — APPROVED** [review]
LGTM!
Could you test it and if it works I'll make a release on this

**Pull request merged**